File: Leetcode/9/9.go
func isPalindrome(x int) bool {
	if x < 0 || (x % 10 == 0 && x != 0) {
		return false
	}
	var xRev = 0
	for x > xRev {
		xRev = xRev*10 + x%10
		x /= 10
	}
	return x == xRev || x == xRev / 10
}

File: Leetcode/9/9.cpp
class Solution {
public:
    bool isPalindrome(int A) {
        if (A < 0) {
            return false;
        }
        if (A % 10 == 0 && A != 0) {
            return false;
        }
        decltype(A) A_rev{0};
        while (A > A_rev) {
            A_rev = A_rev * 10 + A % 10;
            A /= 10;
        }
        return (A == A_rev || A == A_rev / 10);
    }
};
